README.md

```diff
@@ -32,31 +32,418 @@ There's a very unstable version of the plugin available to try out. It's held to
+
+```ts
+type PortalSession = {
+  configuration?: config
+  locale?: string
+  on_behalf_of?: string
+  return_url: string
+}
+```
+
+Set the second parameter, `skipAuth`, to `true` when testing the Customer Portal redirect if you have not yet set up user authentication in your app. Once authentication is set up please remove the argument.
+
+#### `createStripeCustomerPortalConfig(portalConfig)`
+
+Creates a [Customer Portal Configuration](https://stripe.com/docs/api/customer_portal/configuration) object.
+
+```js
+
+```ts
+type PortalConfig = {
+  // TODO
+}
+```
+
+### `<StripeProductsCell params={priceParams, productParams}/>`
+
+As part of the setup process, a `<StripeProductsCell/>` gets generated inside of your app's `components` folder. 
+
+This [Cell](https://redwoodjs.com/docs/cells) returns a list of Products unique to this plugin. It uses a combination of data from [Stripe Prices](https://stripe.com/docs/api/prices) and [Stripe Products](https://stripe.com/docs/api/products). The data is combined as follows: 
+```js
+{
+  id: stripePrice.id,
+  name: stripeProduct.name,
+  description: stripeProduct.description,
+  images: stripeProduct.images,
+  price: stripePrice.unit_amount,
+  type: stripePrice.type,
+}
+```
+
+The `params` passed to the Cell are the same parameters that would be passed to Stripe's Products and Prices list functions. The following example shows how to get a list of Products from `active` Stripe Products that have `recurring` Stripe Prices. In other words, this example shows a Cell that returns a list of active subscriptions.
+
+```js
+import StripeProductsCell from 'src/components/StripeProductsCell/StripeProductsCell'
+
+// ...
+
+<StripeProductsCell
+  params={{
+    productParams: { active: true },
+    priceParams: { type: 'recurring' },
+  }}
+/>
+```
+
+`priceParams` has the following shape (for more information, look at the [Stripe Price list API documentation](https://stripe.com/docs/api/prices/list))
+
+```ts
+type priceParams = {
+  active?: boolean
+  currency?: string
+  product?: string
+  created?: stripeCreated
+  ending_before?: string
+  limit?: number
+  lookup_keys?: string
+  recurring?: stripeRecurringPrice
+  starting_after?: string
+  type?: string
+}
+```
+
+`productParams` has the following shape (for more information, look at the [Stripe Products list API documentation](https://stripe.com/docs/api/products/list))
+
+```ts
+type productParams = {
+  active?: boolean
+  created?: stripeCreated
+  ending_before?: string
+  ids?: string[]
+  limit?: number
+  shippable?: boolean
+  starting_after?: string
+  url?: string
+}
+```
+
+The Cell has it's own UI for illustrative purpose. You should replace the UI with your own.
+
+## Api-side API Reference
+
+### Webhook handling
+
+As part of the setup process, a `/stripeWebhook` function is generated. It uses the plugin handler function to verify and manage webhook events.
```

Comment:
Some might prefer their endpoints be snake case rather than camel case. Maybe throw in a line telling people how to edit that?